develop/2017-05-29-fgrehm-analysis-on-meals-based-on-presence-information.py

```diff
@@ -0,0 +1,563 @@
+        'dataIni={}&dataFim={}&codigoSessao=&parteNomeParlamentar={}&siglaPartido=&siglaUF={}'
+    ).format(
+        meal.issue_date.strftime('%d/%m/%Y'),
+        meal.issue_date.strftime('%d/%m/%Y'),
+        meal.congressperson_name,
+        meal.congressperson_state
+    )
+    return '<a href="{}" target="_blank">{} - {}</a>'.format(
+        url,
+        meal.first_speech_at.strftime('%H:%M:%S'),
+        meal.last_speech_at.strftime('%H:%M:%S')
+    )
+    
+suspects2 = meals_outside_df_during_speeches.copy()
+suspects2.timestamps.fillna('', inplace=True)
+suspects2.receipt_dates.fillna('', inplace=True)
+suspects2['score'] = suspects2.apply(score, axis='columns')
+suspects2['period_in_df'] = suspects2.apply(period_in_df, axis='columns')
+suspects2 = suspects2.query('score > 0').sort_values('score', ascending=False)
+print("Suspicious reimbursements:", len(suspects2))
+
+suspects2 = suspects2[~suspects2.document_id.isin(suspects.document_id)]
+print("Suspicious reimbursements that were not found using presences info:", len(suspects2))
+
+
+# In[20]:
+
+report(suspects2.head(20), [
+    'document_id',
+    'receipt',
+    'issue_date', 
+    'congressperson_name',
+    'score',
+    'company_name',
+    'company_state', 
+    'total_net_value', 
+    'period_in_df'
+])
+
+
+# ## Meals in DF while the congressperson was outside DF
+# 
+# In the past we also found cases where the congressperson was on an official mission outside DF but requested a reimbursement for a meal in DF
+
+# In[21]:
+
+missions = pd.read_csv('../data/2017-06-05-official-missions.xz', dtype={'report_details_link': np.str}, low_memory=False)
+print("Total official mission records:", len(missions))
+
+missions = missions.query('canceled == "No"')
+print("Total official mission records without canceled trips:", len(missions))
+
+missions.participant = missions.participant.apply(lambda x: x.upper())
+missions.start = pd.to_datetime(missions.start, format="%Y-%m-%d").dt.date
+missions.end = pd.to_datetime(missions.end, format="%Y-%m-%d").dt.date
+
+missions = missions.merge(
+    deputies, 
+    left_on=['participant'],
+    right_on=['congressperson_name']
+)
+print("Total official mission records with deputies that have not been canceled:", len(missions))
+
+
+# In[22]:
+
+meals_in_df = meals.query('company_state == "DF"')
+print("Meals in DF:", len(meals_in_df))
+
+
+# In[23]:
+
+suspects3 = []
+for idx, mission in missions.iterrows():
+    meals_while_in_mission = meals_in_df[
+        (meals_in_df.issue_date >= (mission.start + timedelta(days=1))) \
+        & (meals_in_df.issue_date <= (mission.end - timedelta(days=1))) \
+        & (meals_in_df.congressperson_id == mission.congressperson_id)
+    ]
+    if (len(meals_while_in_mission) == 0):
+        continue
+    destination = mission.destination
+    if type(mission.report_details_link) is str:
+        destination = '<a href="{}">{}</a>'.format(mission.report_details_link, destination)
```

Comment:
Same as above, this helps performing manual verification of the report submitted by deputies in case they are available. This helps a lot because sometimes the dates provided [here](http://www.camara.leg.br/missao-oficial/missao-pesquisa?deputado=1&nome-deputado=&nome-servidor=&dati=08%2F03%2F2017&datf=06%2F06%2F2017&nome-evento=) do not match the actual dates that the deputy was away.

For example, http://www.camara.leg.br/missao-oficial/missao-relatorio?codViagem=43314&ponto=811278 says the period was Nov 9th until **Dec** 14th, but if you look at the [report provided](http://www.camara.leg.br/missao-oficial/missao_oficial_Relatorio?codViagem=43314&ponto=811278) it says Nov 9th until _Nov_ 14th

/cc @rodolfo-viana